### PR TITLE
6/more

### LINKED ISSUE #6: add endpoints

### PR DESCRIPTION
**Pull request by pyphilia**
close #6

### CODE REVIEW

**Comment by pyphilia** [comment]
@swouf I've added you as a reviewer. This repository is the interface between the backend and the frontend for the apps. You will see that it simplify a lot the apps and change some namenclature.

File: src/api/apps.ts [hunk 1]
```diff
@@ -1,54 +1,100 @@
...
+  token: string;
+  itemId: string;
+  body: any;
+  apiHost: string;
+}) => {
+  const { token, itemId, apiHost, body } = args;
   return axios
-    .delete(`${API_HOST}/${buildDeleteAppDataRoute({ itemId, id })}`, {
+    .post(`${apiHost}/${buildPostAppDataRoute({ itemId })}`, body, {
+      // data: body,
```
Comment (juancarlosfarah) [inline src/api/apps.ts:44]: Remove this?

File: src/config/utils.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+export const getData = (queryClient: QueryClient) => {
+  const data = queryClient.getQueryData<Record<LocalContext>>(LOCAL_CONTEXT_KEY);
+  const token = queryClient.getQueryData<string>(AUTH_TOKEN_KEY);
+  const itemId = data?.get('itemId');
+  const memberId = data?.get('memberId');
+  return { itemId, memberId, token };
+};
+
+export const getDataOrThrow = (queryClient: QueryClient) => {
```
Comment (juancarlosfarah) [inline src/config/utils.ts:32]: Why don't we just `throw` in the main function?

File: src/config/utils.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
+import { QueryClient } from 'react-query';
+import { Record } from 'immutable';
+import { LocalContext, QueryClientConfig } from '../types';
+import { AUTH_TOKEN_KEY, LOCAL_CONTEXT_KEY } from './keys';
+import { StatusCodes } from 'http-status-codes';
+import { MissingAppIdError, MissingAppOriginError } from './errors';
+export class MissingApiHostError extends Error {
```
Comment (juancarlosfarah) [inline src/config/utils.ts:7]: Space above here.

File: src/config/utils.ts [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+  const token = queryClient.getQueryData<string>(AUTH_TOKEN_KEY);
+  const itemId = data?.get('itemId');
+  const memberId = data?.get('memberId');
+  return { itemId, memberId, token };
+};
+
+export const getDataOrThrow = (queryClient: QueryClient) => {
+  const { itemId, memberId, token } = getData(queryClient);
+  if (!itemId || !memberId || !token) {
+    throw new Error(
```
Comment (juancarlosfarah) [inline src/config/utils.ts:35]: Factor out this error?

**Review by juancarlosfarah — CHANGES_REQUESTED** [review]
Looks good! Just some minor comments.

File: src/config/utils.ts (hunk 2 above)
Comment (pyphilia) [reply src/config/utils.ts:32]: We can, but it was to avoid code duplication.

**Pull request merged**